### PR TITLE
[INTEGRATION][FLINK] create first integration test

### PR DESCRIPTION
**Pull request by pawel-big-lebowski**
Signed-off-by: Pawel Leszczynski <leszczynski.pawel@gmail.com>

### Problem

We need to write the first integration test so that we can easily extend what we already have and make sure things do not get broken.

PR allows finishing #623 issue and contributes to #600

### Solution
 
Integration test can be run with:
```
./gradlew clean build
./gradlew integrationTest
```

The test spins up whole environment including:
 * mockserver container to verify OL requests,
 * zookeeper,
 * kafka & it creates input/output topic & generates few input events,
 * schema-registry,
 * flink jobmanager,
 * flink task manager.

The test ends when jobmanager's log contains info about OpenLineage event sent and verifies an event sent. 

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [ ] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [ ] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [X] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've updated the [`CHANGELOG.md`](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)

### CODE REVIEW

File: integration/flink/src/test/java/io/openlineage/flink/FlinkContainerUtils.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    return genericContainer(network, "confluentinc/cp-schema-registry:6.2.1", "schema-registry")
+        .withExposedPorts(28081)
+        .withEnv("SCHEMA_REGISTRY_KAFKASTORE_BOOTSTRAP_SERVERS", "PLAINTEXT://kafka:9092")
+        .withEnv("SCHEMA_REGISTRY_HOST_NAME", "schema-registry")
+        .withEnv("SCHEMA_REGISTRY_LISTENERS", "http://schema-registry:8081,http://0.0.0.0:28081")
+        .dependsOn(startable);
+  }
+
+  static GenericContainer<?> makeKafkaContainer(Network network, Startable zookeeper) {
+    return genericContainer(network, "confluentinc/cp-kafka:6.2.1", "kafka")
```
Comment (mobuchowski) [inline integration/flink/src/test/java/io/openlineage/flink/FlinkContainerUtils.java:40]: Can we use other Kafka image, such as `wurstmeister/kafka:2.13-2.8.1`? Confluent images don't work on ARM.

File: integration/flink/src/test/java/io/openlineage/flink/FlinkContainerUtils.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        .withEnv("KAFKA_BROKER_ID", "1")
+        .withEnv(
+            "KAFKA_LOG4J_LOGGERS",
+            "kafka.controller=INFO,kafka.producer.async.DefaultEventHandler=INFO,state.change.logger=INFO")
+        .dependsOn(zookeeper);
+  }
+
+  @SneakyThrows
+  static GenericContainer<?> makeInitKafkaContainer(Network network, Startable kafka) {
+    return genericContainer(network, "confluentinc/cp-kafka:6.2.1", "init-kafka")
```
Comment (mobuchowski) [inline integration/flink/src/test/java/io/openlineage/flink/FlinkContainerUtils.java:64]: Also, would be nice to have image names in static variables, so we can change the versions in one place.

File: integration/flink/src/test/java/io/openlineage/flink/FlinkContainerUtils.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    return genericContainer(network, "confluentinc/cp-zookeeper:6.2.1", "zookeeper")
+        .withExposedPorts(2181)
+        .withEnv("ZOOKEEPER_CLIENT_PORT", "2181")
+        .withEnv("ZOOKEEPER_SERVER_ID", "1");
+  }
+
+  static GenericContainer<?> makeFlinkJobManagerContainer(
+      Network network, List<Startable> startables) {
+    GenericContainer<?> container =
+        genericContainer(network, "flink:1.14.2-scala_2.11-java11", "jobmanager")
```
Comment (mobuchowski) [inline integration/flink/src/test/java/io/openlineage/flink/FlinkContainerUtils.java:99]: Let's use Scala 2.12 when possible.

File: integration/flink/src/test/java/io/openlineage/flink/OpenLineageFlinkContainerTest.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        .when(request("/api/v1/lineage"))
+        .respond(org.mockserver.model.HttpResponse.response().withStatusCode(201));
+
+    Awaitility.await().until(openLineageClientMockContainer::isRunning);
+  }
+
+  @AfterAll
+  public static void tearDown() {
+    try {
+      openLineageClientMockContainer.stop();
```
Comment (mobuchowski) [inline integration/flink/src/test/java/io/openlineage/flink/OpenLineageFlinkContainerTest.java:81]: Maybe separate method that stops passed image?

**Review by mobuchowski — CHANGES_REQUESTED** [review]

File: integration/flink/src/test/java/io/openlineage/flink/FlinkContainerUtils.java (hunk 1 above)
Comment (pawel-big-lebowski) [reply integration/flink/src/test/java/io/openlineage/flink/FlinkContainerUtils.java]: please, check it on ARM then ;-)

**Review by mobuchowski — APPROVED** [review]

**Pull request merged**